osbrain/tests/test_proxy.py

```diff
@@ -119,6 +122,48 @@ def ping(self):
         InitTimeoutProxy('foo', timeout=1.)
 
 
+@pytest.mark.parametrize('timeout', [-.5, 0., .5])
+def test_agent_proxy_wait_running(nsproxy, timeout):
+    """
+    Using `wait_for_running` on a proxy after initialization should block until
+    the agent is running or time out.
+    """
+    ap = AgentProcess('agent')
+    ap.start()
+
+    # Get "offline" proxy
+    agent = Proxy('agent')
+    time0 = time.time()
+    Timer(abs(timeout)/2, agent.run).start()
+
+    # Wait long enough for the agent to start running
+    proxy = Proxy('agent').wait_for_running(timeout=timeout)
+    assert proxy.ping() == 'pong'
+    if timeout != 0:
+        assert abs(timeout)/2 <= time.time() - time0 <= abs(timeout)
+
+
+# Can't use negatives here because they will never time out
+@pytest.mark.parametrize('timeout', [0., .5])
+def test_agent_proxy_wait_running_timeout(nsproxy, timeout):
+    """
+    Check that the `wait_for_running` method times out if the agent is not
+    running after the specified number of seconds.
+    """
+    ap = AgentProcess('agent')
+    ap.start()
```

Comment:
You are not using `ap` variable, so maybe you can `AgentProcess('agent').start()` instead? (one line, no variable).

Reply:
True. I probably was using it before, and that's why it was left here.